tests/unit/test_handlers/test_http_handlers/test_options.py

```diff
@@ -0,0 +1,27 @@
+import pytest
+
+from litestar import Router, get
+from litestar.testing import create_test_client
+
+
+@pytest.mark.xfail(reason="broken behaviour that never really worked")
+def test_option_handler_inherits_layer_config() -> None:
+    # Currently broken. The reason this cannot work reliably is that when auto-creating
+    # OPTIONS handlers, we *assume* that it should inherit the configuration of
+    # path-adjacent route handlers, e.g. if a route handler is defined on '/', and
+    # registered on a router with the path '/one', it would receive the configuration
+    # from that router. However, since it is possible to have multiple handlers per
+    # path, which can also be registered with different routers, we would still only
+    # create one 'OPTIONS' handler for each path, even if multiple routers exist with
+    # handlers for that path. In this case, it is unclear which configuration the
+    # 'OPTIONS' handler should receive
```

Comment:
I trust you on this , but cant we defer OPTIONS creation once the tree is complete ? maybe it is already the case I admit the flow inside my brain isnt yet as clear as Id like

Reply:
We could, but would that solve the problem of it not working as expected?

After we've resovled everything, we don't have the context of the layers anymore at all, so we have even less information to go on. Not sure if that's better.

Although I guess you *could* make the case that we should just "copy" the handler config from the adjacent routes, but then there's the question again of which one to use..